### ISSUE TITLE
Calendar widget (jQuery UI DatePicker) with numberOfMonths = 2 or more

### ISSUE THREAD

**Issue by atihomirov** [magento/magento2#7379]
### Preconditions

1.  Magento version 2.1.1 
2.  file \lib\web\mage\calendar.js

### Steps to reproduce

1.  set options to widget  'numberOfMonths'=> array(1,3),
2.   set options on widget to any non-default values for 'numberOfMonths'

### Expected result

1.  jQuery UI DatePicker Widget with a list of months

### Actual result

1. the first month is normal - the remaining corrupted
2. month header repeat
3. next month days not show

change:
add line 248:  "calender = '';
reinit loop values
line 277: for (dow=0;... 
line 295: for (dRow=0;...

**Comment by tkacheva** [comment]
@atihomirov Thanks for reporting! please explain the business case you try to achieve here.

**Comment by atihomirov** [comment]
using extension "Amasty delivery date" the client needs to display multiple months for clarity.
the code for the call:
```
$elements['deliverydate_date'] = [
                    'component' => 'Amasty_Deliverydate/js/checkout/date',
...
                    'config' => [
...
                        'options' => ['amdeliveryconf' => $this->deliveryHelper->getDeliveryConfig(),
                                      'numberOfMonths'=> array(1,2),  ],
                    ]
                ];
```
or, for reproduce:  in file  `\lib\web\mage\calendar.js` change
```
line 127:                    navigationAsDateFormat = this._get(inst, 'navigationAsDateFormat'),
line 128:                    numMonths = [1,2],
line 129:                    showCurrentAtPos = this._get(inst, 'showCurrentAtPos'),
```

могу подробнее описать на русском, если нужно.

**Comment by tkacheva** [comment]
@atihomirov I appreciate the actual result screenshot as well. The rest is clear.

**Comment by atihomirov** [comment]
![sgc_450_actual](https://cloud.githubusercontent.com/assets/1183236/20389110/b13d2a72-acd0-11e6-8c00-1eb10977c85b.png)
![sgc_450_shoudby](https://cloud.githubusercontent.com/assets/1183236/20389111/b140a1e8-acd0-11e6-81df-2b74de57f7ff.png)

**Comment by atihomirov** [comment]
1 image 
if not set calendar=""; - Widget shows the calendar text in the recursion (november 1, november 2 + december )
if  not set dRow=0; dow=0;  - Widget not show deсember dates because the dates have become 31 and would never again 1 of the month

**Comment by tkacheva** [comment]
issue MAGETWO-61209

**Comment by magento-engcom-team** [comment]
@atihomirov, thank you for your report.
We've created internal ticket(s) MAGETWO-61209 to track progress on the issue.

**Comment by sidolov** [comment]
Hi @atihomirov. Thank you for your report. 
The issue has been fixed in magento/magento2#16278 by @vasilii-b in 2.3-develop branch
Related commit(s):
 - [54e1a715be3ef0bdfdb45b6fde99912d6e19fca6](https://github.com/magento/magento2/commit/54e1a715be3ef0bdfdb45b6fde99912d6e19fca6)
 - [e874ff0cb963bdb2a700c35c74b6051318705a5b](https://github.com/magento/magento2/commit/e874ff0cb963bdb2a700c35c74b6051318705a5b)

The fix will be available with the upcoming 2.3.0 release.

**Comment by sidolov** [comment]
Hi @atihomirov. Thank you for your report. 
The issue has been fixed in magento/magento2#16280 by @vasilii-b in 2.1-develop branch
Related commit(s):
 - [ce38bc87478f629ed7158df526f1977f2e3d76ef](https://github.com/magento/magento2/commit/ce38bc87478f629ed7158df526f1977f2e3d76ef)

The fix will be available with the upcoming 2.1.15 release.

**Comment by magento-engcom-team** [comment]
Hi @atihomirov. Thank you for your report. 
The issue has been fixed in magento/magento2#16279 by @vasilii-b in 2.2-develop branch
Related commit(s):
 - [de533cca86414906fc04302a7b7e3ec92186d85a](https://github.com/magento/magento2/commit/de533cca86414906fc04302a7b7e3ec92186d85a)

The fix will be available with the upcoming 2.2.6 release.
